### PR TITLE
Dt 963 search phrase

### PR DESCRIPTION
**Pull request by andymarke**
- add ability to seed all data we want to test
- add single test to prove seeding works
- add single basic search to prove we get some sort of results

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,6 +25,13 @@ jobs:
     executor: validator
     steps:
       - checkout
+      - hmpps/install_aws_cli
+      - run:
+          name: Wait for ES to be ready
+          command: curl -4 --connect-timeout 30 --retry-connrefused --retry 20 --retry-delay 5 http://localhost:4578
+      - run:
+          name: Set up es domain
+          command: src/test/resources/localstack/setup-es.sh
```
Comment (andymarke) [inline .circleci/config.yml:34]: moving to localstack 11 means "manually" creating ES domain

File: build.gradle.kts [hunk 2]
```diff
@@ -29,14 +29,13 @@ dependencies {
...
   implementation("org.apache.commons:commons-text:1.9")
 
-  implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.11.1")
+  implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.11.2")
 
-  // elasticsearch version must match the aws cluster version (6.7)
-  implementation("org.elasticsearch:elasticsearch:6.7.2")
-  implementation("org.elasticsearch.client:elasticsearch-rest-high-level-client:6.7.2")
-  implementation("org.elasticsearch.client:elasticsearch-rest-client:6.7.2")
+  implementation("org.elasticsearch:elasticsearch:7.4.1")
```
Comment (andymarke) [inline build.gradle.kts:34]: the breaking change in the sense this only works against ES version 7.x) therefore will only work against cloud platform version of ES (Delius AWS has version 6.x)

File: src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchControllerPhraseTest.kt [hunk 3]
```diff
@@ -329,11 +328,11 @@ class OffenderSearchControllerPhraseTest {
 
   @Test
   internal fun `will return suggestions`() {
-    val suggestion: Suggestion<Entry<Option>> = Suggestion<Entry<Option>>("firstName", 5).apply {
-      this.addTerm(Entry<Option>(Text("smyth"), 0, 5).apply {
-        this.addOption(Option(Text("smith"), 0.8f))
-        this.addOption(Option(Text("smita"), 0.8f))
-        this.addOption(Option(Text("sumith"), 0.8f))
+    val suggestion: TermSuggestion = TermSuggestion("firstName", 5, SCORE).apply {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchControllerPhraseTest.kt:331]: due to ES v7 API changes

File: src/test/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchServiceTest.kt [hunk 4]
```diff
@@ -563,7 +565,7 @@ internal class MatchServiceTest {
 
   private fun resultsOf(vararg offenders: OffenderDetail): SearchResponse {
     val searchHits = offenders.map { SearchHit(it.offenderId.toInt()).apply { sourceRef(BytesArray(objectMapper.writeValueAsBytes(it))) } }
-    val hits = SearchHits(searchHits.toTypedArray(), offenders.size.toLong(), 10f)
+    val hits = SearchHits(searchHits.toTypedArray(), TotalHits(offenders.size.toLong(), EQUAL_TO), 10f)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchServiceTest.kt:568]: due ES v7 changes

File: src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchPhraseAPIIntegrationTest.kt [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+import java.time.LocalDate
+import java.util.*
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles(profiles = ["test", "localstack"])
+@ExtendWith(SpringExtension::class)
+@TestExecutionListeners(listeners = [DependencyInjectionTestExecutionListener::class, OffenderSearchPhraseAPIIntegrationTest::class])
+@ContextConfiguration
+@DirtiesContext
+class OffenderSearchPhraseAPIIntegrationTest : AbstractTestExecutionListener() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchPhraseAPIIntegrationTest.kt:41]: Can we pare this back to the minimal spring boot test config - not sure if all the complexity is needed?

**Pull request merged**